controllers/perconaservermysql_controller.go

```diff
@@ -148,6 +152,75 @@ func (r *PerconaServerMySQLReconciler) doReconcile(
 	return nil
 }
 
+func (r *PerconaServerMySQLReconciler) reconcileVersions(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL) error {
+	if cr.Spec.UpgradeOptions.Apply == "" ||
+		cr.Spec.UpgradeOptions.Apply == apiv1alpha1.UpgradeStrategyDisabled ||
+		cr.Spec.UpgradeOptions.Apply == apiv1alpha1.UpgradeStrategyNever {
+		return nil
+	}
+
+	l := log.FromContext(ctx).WithName("reconcileVersions")
+
+	version, err := vs.GetVersion(ctx, cr, r.ServerVersion)
+	if err != nil {
+		return errors.Wrap(err, "get exact version")
+	}
+
+	patch := client.MergeFrom(cr.DeepCopy())
+	if cr.Spec.MySQL.Image != version.PSImage {
+		if cr.Status.MySQL.Version == "" {
+			l.Info("set MySQL version to " + version.PSVersion)
+		} else {
+			l.Info("update MySQL version", "old version", cr.Status.MySQL.Version, "new version", version.PSVersion)
+		}
+		cr.Spec.MySQL.Image = version.PSImage
+	}
+	if cr.Spec.Backup.Image != version.BackupImage {
+		if cr.Status.BackupVersion == "" {
+			l.Info("set backup version to " + version.BackupVersion)
+		} else {
+			l.Info("update backup version", "old version", cr.Status.BackupVersion, "new version", version.BackupVersion)
+		}
+		cr.Spec.Backup.Image = version.BackupImage
+	}
+	if cr.Spec.Orchestrator.Image != version.OrchestratorImage {
+		if cr.Status.Orchestrator.Version == "" {
+			l.Info("set orchestrator version to " + version.OrchestratorVersion)
+		} else {
+			l.Info("update orchestrator version", "old version", cr.Status.Orchestrator.Version, "new version", version.OrchestratorVersion)
+		}
+		cr.Spec.Orchestrator.Image = version.OrchestratorImage
+	}
+	if cr.Spec.Router.Image != version.RouterImage {
+		if cr.Status.Router.Version == "" {
+			l.Info("set MySQL router version to " + version.RouterVersion)
+		} else {
+			l.Info("update MySQL router version", "old version", cr.Status.Router.Version, "new version", version.RouterVersion)
+		}
+		cr.Spec.Router.Image = version.RouterImage
+	}
+	if cr.Spec.PMM.Image != version.PMMImage {
+		if cr.Status.PMMVersion == "" {
+			l.Info("set PMM version to " + version.PMMVersion)
+		} else {
+			l.Info("update PMM version", "old version", cr.Status.PMMVersion, "new version", version.PMMVersion)
+		}
+		cr.Spec.PMM.Image = version.PMMImage
+	}
+
+	err = r.Patch(ctx, cr.DeepCopy(), patch)
+	if err != nil {
+		return errors.Wrap(err, "failed to update CR")
+	}
+
+	cr.Status.MySQL.Version = version.PSVersion
+	cr.Status.BackupVersion = version.BackupVersion
+	cr.Status.Orchestrator.Version = version.OrchestratorVersion
+	cr.Status.Router.Version = version.RouterVersion
+	cr.Status.PMMVersion = version.PMMVersion
```

Comment:
shouldn't we update status here?

Reply:
Status is updated at the end of `Reconcile` loop https://github.com/percona/percona-server-mysql-operator/blob/7e5553b4bfc7411d706810ef1b770d6029d00a28/controllers/perconaservermysql_controller.go#L106-L110 Is it ok?